src/App.tsx

```diff
@@ -1,14 +1,56 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo } from './types/Todo';
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState<string>('all');
+
+  const filterTodos = (
+    todosArray: Todo[],
+    searchQuery: string,
+    currentStatus: string,
+  ) => {
+    return todosArray
+      .filter(todo =>
+        todo.title.toLowerCase().includes(searchQuery.trim().toLowerCase()),
+      )
+      .filter(todo => {
+        switch (currentStatus) {
+          case 'completed':
+            return todo.completed;
+
+          case 'active':
+            return !todo.completed;
+
+          case 'all':
+          default:
+            return true;
+        }
+      });
+  };
+
+  const filteredTodos = filterTodos(todos, query, status);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => {
+        setLoading(false);
+      });
```

Comment:
Consider adding error handling to the `getTodos()` promise chain. This will help manage any potential errors that occur during the fetching of todos. You can use a `.catch()` block to handle errors.